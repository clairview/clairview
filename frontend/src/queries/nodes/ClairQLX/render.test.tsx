import { parseClairQLX, renderClairQLX } from '~/queries/nodes/ClairQLX/render'

describe('ClairQLX', () => {
    describe('parse', () => {
        it('should parse tags', () => {
            const value = parseClairQLX(['__hx_tag', 'Sparkline', 'data', [1, 2, 3], 'type', ['line']])
            expect(value).toEqual({
                __hx_tag: 'Sparkline',
                data: [1, 2, 3],
                type: ['line'],
            })
        })

        it('should parse empty tags', () => {
            const value = parseClairQLX(['__hx_tag', 'Sparkline'])
            expect(value).toEqual({
                __hx_tag: 'Sparkline',
            })
        })

        it('should parse objects', () => {
            const value = parseClairQLX(['__hx_tag', '__hx_obj', 'a', 1, 'b', 2])
            expect(value).toEqual({
                a: 1,
                b: 2,
            })
        })

        it('should handle arrays', () => {
            const value = parseClairQLX(['a', 'b', 'c'])
            expect(value).toEqual(['a', 'b', 'c'])
        })

        it('should handle nested arrays', () => {
            const value = parseClairQLX(['a', ['b', 'c']])
            expect(value).toEqual(['a', ['b', 'c']])
        })

        it('should handle nested objects', () => {
            const value = parseClairQLX(['__hx_tag', '__hx_obj', 'a', ['b', 'c']])
            expect(value).toEqual({
                a: ['b', 'c'],
            })
        })

        it('should handle nested objects with tags', () => {
            const value = parseClairQLX([
                '__hx_tag',
                '__hx_obj',
                'a',
                ['__hx_tag', 'Sparkline', 'data', [1, 2, 3], 'type', ['line']],
            ])
            expect(value).toEqual({
                a: {
                    __hx_tag: 'Sparkline',
                    data: [1, 2, 3],
                    type: ['line'],
                },
            })
        })
    })

    describe('render', () => {
        it('should render Sparkline', () => {
            const value = {
                __hx_tag: 'Sparkline',
                data: [1, 2, 3],
                type: ['line'],
            }
            const element = renderClairQLX(value)
            expect(element).toMatchSnapshot()
        })

        it('should render object', () => {
            const value = {
                a: 1,
                b: 2,
            }
            const element = renderClairQLX(value)
            expect(element).toMatchSnapshot()
        })

        it('should render unknown tag', () => {
            const value = {
                __hx_tag: 'Unknown',
            }
            const element = renderClairQLX(value)
            expect(element).toMatchSnapshot()
        })

        it('should render array', () => {
            const value = [1, 2, 3]
            const element = renderClairQLX(value)
            expect(element).toMatchSnapshot()
        })
    })
})
