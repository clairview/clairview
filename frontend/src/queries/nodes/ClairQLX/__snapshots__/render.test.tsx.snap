// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClairQLX render should render Sparkline 1`] = `
<ErrorBoundary>
  <Sparkline
    className="h-8"
    data={
      [
        1,
        2,
        3,
      ]
    }
    type={
      [
        "line",
      ]
    }
  />
</ErrorBoundary>
`;

exports[`ClairQLX render should render array 1`] = `
<JSONViewer
  collapsed={1}
  name={null}
  src={
    [
      1,
      2,
      3,
    ]
  }
/>
`;

exports[`ClairQLX render should render object 1`] = `
<JSONViewer
  collapsed={1}
  name={null}
  src={
    {
      "a": 1,
      "b": 2,
    }
  }
/>
`;

exports[`ClairQLX render should render unknown tag 1`] = `
<div>
  Unknown tag: 
  Unknown
</div>
`;
